raw_data_QA/check_nyt_raw_data.py

```diff
@@ -0,0 +1,508 @@
+import pandas as pd
+import matplotlib.pyplot as plt
+import numpy as np
+from sklearn.metrics import mean_squared_error
+from datetime import datetime
+import calendar
+import argparse
+import pdb
+import os
+import shutil
+
+
+def aggregate_df(df, args):
+    # get all county level data points and add to get state level data points
+    aggregation_functions = {
+        args.new_cases_name: "sum",
+        args.new_deaths_name: "sum",
+        args.date_name: "first",
+    }
+    # add up all entries with same date
+    df_new = df.groupby(df[args.date_name]).aggregate(aggregation_functions)
+    # Get the number of new cases by taking the difference of cumulative cases given by NYT
+    df_new["new_cases"] = df_new[args.new_cases_name].diff().fillna(0)
+    df_new["new_deaths"] = df_new[args.new_deaths_name].diff().fillna(0)
+    return df_new
+
+
+def get_state(df, state, args):
+    df_new = df.loc[df[args.state_name] == state]
+    return df_new
+
+
+def get_rmse(df1, df2, var):
+    rmse = np.sqrt(mean_squared_error(df1[[var]], df2[[var]]))
+    return round(rmse, 2)
+
+
+def get_equal_len_df(df1, df2):
+    df2_length = len(df2.index)
+    truncated_df1 = df1.head(df2_length)
+    return truncated_df1
+
+
+def get_p_diff_z_score(var1, var2):
+    if var1 == 0 and var2 == 0:
+        p_diff = 0
+        z_score = 0
+    elif var1 == 0 and var2 != 0:  # may want to deal with small variations differently
+        p_diff = (var2 - var1) / var2
+        z_score = (var2 - var1) / 1  # I am hardcoding the error on zero to be 1
+    elif var2 == 0 and var1 != 0:
+        p_diff = (var2 - var1) / var1
+        z_score = (var2 - var1) / 1  # again hardcoding error on zero to be 1
+    else:
+        p_diff = (abs(var2) - abs(var1)) / var1
+        err = np.sqrt(abs(var1))
+        z_score = (var2 - var1) / err  # to be added
+    return p_diff, z_score
+
+
+def get_compare_metrics(df1, df2, var):
+    diff = []
+    z_scores = []
+    for i in range(len(df1.index)):
+        var1 = df1[var].iloc[i]
+        var2 = df2[var].iloc[i]
+        p_diff, z_score = get_p_diff_z_score(var1, var2)
+
+        z_scores.append(z_score)
+        diff.append(p_diff)
+        number_of_days_percent_threshold = sum(i > args.percent_threshold for i in diff)
+        average_Z = round(average(z_scores), 2)
+        latest_Z = round(average(z_scores[: -args.n_days_z_score_mean]), 2)
+    return diff, z_scores, average_Z, latest_Z, number_of_days_percent_threshold
+
+
+def average(list):
```

Comment:
maybe you answered this in the other PR, but any reason to define this function rather than using something in `np`?

Reply:
Fixed